### PR TITLE
docs/annotations-gui

### PR DESCRIPTION
**Pull request by bm64**
Extracted build custom UI section from stock tools docs and created annotations GUI docs.

### CODE REVIEW

**Review by pawellysy — APPROVED** [review]

File: docs/advanced-chart-features/annotations-gui.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+2. Hook up the functions for the events for [`navigation.bindings`](https://api.highcharts.com/highcharts/navigation.bindings).
+
+  ```js
+  navigation: {
+      // Informs Annotations module where to look for HTML elements for adding annotations etc.
+      bindingsClassName: 'custom-gui-container',
+      events: {
+          // On selecting the annotation the showPopup event is fired
+          showPopup: function(event) {
+              var chart = this.chart;
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:64]:
```suggestion
              const chart = this.chart;
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+_The code snippet below demonstrates how trigger lookup of a custom GUI with the `navigation.bindingsClassName`._
+
+```js
+Highcharts.chart('container', {
+    navigation: {
+        bindingsClassName: 'custom-gui-container' // informs Annotations module where to look for HTML elements for adding annotations etc.
+    },
+    series: [{
+        ....
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:18]:
```suggestion
        ...
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+_The code snippet below demonstrates how trigger lookup of a custom GUI with the `navigation.bindingsClassName`._
+
+```js
+Highcharts.chart('container', {
+    navigation: {
+        bindingsClassName: 'custom-gui-container' // informs Annotations module where to look for HTML elements for adding annotations etc.
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:15]:
```suggestion
        bindingsClassName: 'custom-gui-container' // Informs Annotations module where to look for HTML elements for adding annotations etc.
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+### Binding an event to a HTML element
+HTML elements are bound to annotations events through reverse lookup of HTML elements with a classname in the form of: `highcharts-BINDINGNAME`. For example a HTML button with `highcharts-circle-annotation` will bind the [`navigation.bindings.circleAnnotation`](https://api.highcharts.com/highcharts/navigation.bindings.circleAnnotation) events. Class name can be changed by setting `navigation.bindings.circleAnnotation.className`. The actual binding takes place on chart initialization, but only for HTML elements that are wrapped within another HTML element with the classname correlating the chart configuration for [`navigation.bindingsClassName`](https://api.highcharts.com/highcharts/navigation.bindingsClassName). The wrapping classname defaults to `highcharts-bindings-wrapper`. Wrapping the GUI elements is made required for specifying which GUI elements need to be bound and events are delegated to the correct chart.
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:8]:
```suggestion
HTML elements are bound to annotations events through reverse lookup of HTML elements with a classname in the form of: `highcharts-BINDINGNAME`. For example a HTML button with `highcharts-circle-annotation` will bind the [`navigation.bindings.circleAnnotation`](https://api.highcharts.com/highcharts/navigation.bindings.circleAnnotation) events. Classname can be changed by setting `navigation.bindings.circleAnnotation.className`. The actual binding takes place on chart initialization, but only for HTML elements that are wrapped within another HTML element with the classname correlating the chart configuration for [`navigation.bindingsClassName`](https://api.highcharts.com/highcharts/navigation.bindingsClassName). The wrapping classname defaults to `highcharts-bindings-wrapper`. Wrapping the GUI elements is made required for specifying which GUI elements need to be bound and events are delegated to the correct chart.
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+3. Add an event listener to the "Save" button of the dialog window for saving the color to the `fill` of the circle annotation.
+
+  ```js
+  Highcharts.chart('container', {
+      chart: {
+          events: {
+              load: function() {
+                  // Function which saves the new background color.
+                  var chart = this;
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:111]:
```suggestion
                  const chart = this;
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                  // Select the save button of the popup and assign a click event
+                  document.querySelectorAll('.highcharts-popup-annotations button')[0].addEventListener(
+                      'click',
+                      function() {
+                          var color = document.querySelectorAll(
```
Comment (kamil-musialowski) [inline docs/advanced-chart-features/annotations-gui.md:116]:
```suggestion
                          const color = document.querySelectorAll(
```

File: samples/highcharts/annotations/basic-gui/demo.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+Highcharts.chart('container', {
+    navigation: {
+        // informs Annotations module where to look for HTML elements for adding
```
Comment (kamil-musialowski) [inline samples/highcharts/annotations/basic-gui/demo.js:3]:
```suggestion
        // Informs Annotations module where to look for HTML elements for adding
```

**Review by kamil-musialowski — CHANGES_REQUESTED** [review]

**Review by kamil-musialowski — APPROVED** [review]
lgtm!

**Review by DJTechnoo — APPROVED** [review]
Great job!

File: docs/advanced-chart-features/annotations-gui.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+### Binding an event to a HTML element
+HTML elements are bound to annotations events through reverse lookup of HTML elements with a classname in the form of: `highcharts-BINDINGNAME`. For example a HTML button with `highcharts-circle-annotation` will bind the [`navigation.bindings.circleAnnotation`](https://api.highcharts.com/highcharts/navigation.bindings.circleAnnotation) events. Classname can be changed by setting `navigation.bindings.circleAnnotation.className`. The actual binding takes place on chart initialization, but only for HTML elements that are wrapped within another HTML element with the classname correlating the chart configuration for [`navigation.bindingsClassName`](https://api.highcharts.com/highcharts/navigation.bindingsClassName). The wrapping classname defaults to `highcharts-bindings-wrapper`. Wrapping the GUI elements is made required for specifying which GUI elements need to be bound and events are delegated to the correct chart.
```
Comment (TorsteinHonsi) [inline docs/advanced-chart-features/annotations-gui.md:8]:
```suggestion
HTML elements are bound to annotations events through reverse lookup of HTML elements with a class name in the form of: `highcharts-BINDINGNAME`. For example a HTML button with `highcharts-circle-annotation` will bind the [`navigation.bindings.circleAnnotation`](https://api.highcharts.com/highcharts/navigation.bindings.circleAnnotation) events. Class name can be changed by setting `navigation.bindings.circleAnnotation.className`. The actual binding takes place on chart initialization, but only for HTML elements that are wrapped within another HTML element with the class name correlating the chart configuration for [`navigation.bindingsClassName`](https://api.highcharts.com/highcharts/navigation.bindingsClassName). The wrapping class name defaults to `highcharts-bindings-wrapper`. Wrapping the GUI elements is made required for specifying which GUI elements need to be bound and events are delegated to the correct chart.
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    navigation: {
+        bindingsClassName: 'custom-gui-container' // Informs Annotations module where to look for HTML elements for adding annotations etc.
+    },
+    series: [{
+        ...
+    }]
+});
+```
+
+_The button below is now bound to the `circleAnnotation` events, after it's found through the above specified classname `custom-gui-container`._
```
Comment (TorsteinHonsi) [inline docs/advanced-chart-features/annotations-gui.md:23]:
```suggestion
_The button below is now bound to the `circleAnnotation` events, after it's found through the above specified class name `custom-gui-container`._
```

File: docs/advanced-chart-features/annotations-gui.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+<iframe style="width: 100%; height: 469px; border: none;" src=https://www.highcharts.com/samples/embed/highcharts/annotations/basic-gui allow="fullscreen"></iframe>
+
+### Dialog windows
+
+In applications with more advanced user interaction, additional dialog windows are likely needed. Support for building custom dialog windows is facilitated by hooking up functions to the [`navigation.bindings`](https://api.highcharts.com/highcharts/navigation.bindings).
+
+How to implement a custom dialog window, is best explained with an example where we change the background color of a circle annotation.
+
+1. First we define a dialog window in HTML. Notice the classname `highcharts-popup-annotations` of the outer `div` element. We need this classname in the [`navigation.events.showPopup`](https://api.highcharts.com/highcharts/navigation.events.showPopup) event later on.
```
Comment (TorsteinHonsi) [inline docs/advanced-chart-features/annotations-gui.md:41]:
```suggestion
1. First we define a dialog window in HTML. Notice the class name `highcharts-popup-annotations` of the outer `div` element. We need this class name in the [`navigation.events.showPopup`](https://api.highcharts.com/highcharts/navigation.events.showPopup) event later on.
```

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, looks good! Suggested correcting "classname" to "class name".

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**